tools/cooja/apps/avrora/build.gradle

```diff
@@ -0,0 +1,24 @@
+apply plugin: "java"
+
+jar.baseName = 'cooja_avrora'
+
+repositories {
+  jcenter()
+}
+
+sourceSets {
+  main {
+    java {
+      srcDir 'src'
+    }
+  }
+
+}
+
+
+dependencies {
+    compile project(":")
+    compile files("lib/avrora-small.jar")
```

Comment:
Depending on relative paths of local files for your Gradle build is really bad practice. You should instead install them in a local repository (I recommend using a local Maven repo - don't worry, nothing to do with Maven, just the structure of the files) if they are not part of the same build:

```groovy
repositores {
    mavenLocal()
    jcenter()
}
```

All jars will then be installed at `${user.home}/.m2/repository` (by default). You don't need to know that, though, as Gradle will make its magic from there.

Change all dependencies you have of this type to be regular dependencies of the form `compile "group:name:version"`. 

Of course, each project has to be given a group and version (name is the name of the directory unless you specify it).

Run the build with `gradle clean publishToMavenLocal`.

You can later even publish all your artifacts (jars) to bintray (JCenter) if you want to so people can depend on your "jars" in their projects too.

Regarding the dependency on the root project (`compile project(':')`), try replacing this line with the following

```groovy
compile rootProject
```

This tells Gradle your child project has a dependency on the root project.

But as I told you in the Java Meetup, think of the build as a tree of projects where a project cannot depend on another project which is higher in the same branch of the tree, they can depend only on "siblings".

Please have a look at some of my projects, I have a really interesting one with multiple projects built together [here](https://github.com/renatoathaydes/osgi-run/tree/master/osgi-run-test).